recipe/meta.yaml

```diff
@@ -85,71 +87,172 @@ requirements:
+        - {{ cdt('libxrender-devel') }}      # [linux]
+        - {{ cdt('libxxf86vm') }}            # [linux]
+        - {{ cdt('mesa-dri-drivers') }}      # [linux]
+        - {{ cdt('mesa-libgl-devel') }}      # [linux]
+        - {{ cdt('xorg-x11-proto-devel') }}  # [linux]
+      files:
+        - test-cmake/CMakeLists.txt
+        - test-cmake/DisplayImage.cpp
+        - test.cpp
+      commands:
+        - pushd test-cmake
+        - cmake . -GNinja -DOpenCV_SHARED=ON -G"Visual Studio 14 2015"        # [win32]
+        - cmake . -GNinja -DOpenCV_SHARED=ON -G"Visual Studio 14 2015 Win64"  # [win64]
+        - cmake . -GNinja -DOpenCV_SHARED=ON                                  # [not win]
+        - cmake --build . --config Release
+        - popd
+        # Verify dynamic libraries.
+        # "bioinspired", Not working in 3.1.0
+        {% set opencv_libs = [
+             "aruco",
+             "bgsegm",
+             "calib3d",
+             "ccalib",
+             "core",
+             "datasets",
+             "dpm",                               # [x86_64]
+             "face",                              # [X86_64]
+             "features2d",
+             "flann",
+             "fuzzy",
+             "highgui",
+             "imgcodecs",
+             "imgproc",
+             "line_descriptor",
+             "ml",
+             "objdetect",                        # [x86_64]
+             "optflow",
+             "phase_unwrapping",
+             "photo",
+             "plot",
+             "reg",
+             "rgbd",
+             "saliency",
+             "shape",
+             "stereo",
+             "stitching",
+             "structured_light",
+             "superres",
+             "surface_matching",
+             "text",                             # [x86_64]
+             "tracking",
+             "video",
+             "videoio",
+             "videostab",
+             "xfeatures2d",
+             "ximgproc",
+             "xobjdetect",                       # [x86_64]
+             "xphoto",
+        ] %}
+        {% for each_opencv_lib in opencv_libs %}
+        - test -f $PREFIX/lib/libopencv_{{ each_opencv_lib }}.dylib                        # [osx]
+        - test -f $PREFIX/lib/libopencv_{{ each_opencv_lib }}.so                           # [linux]
+        - if not exist %PREFIX%\\Library\\bin\\opencv_{{ each_opencv_lib }}342.dll exit 1  # [win]
+        {% endfor %}
+
+  - name: libopencv
+    script: install-nothing.sh
+    version: {{ version }}
+    build:
+      number: {{ build_num }}
+      skip: True  # [(win and py27) or s390x]
+    requirements:
+      host:
+        - {{ pin_subpackage('opencv', max_pin='x') }}
+
+  - name: py-opencv
+    script: install-nothing.sh
+    version: {{ version }}
+    build:
+      number: {{ build_num }}
+      skip: True  # [(win and py27) or s390x]
+    requirements:
+      host:
+        - {{ pin_subpackage('opencv', max_pin='x') }}
```

Comment:
For the python output `py-opencv` please add explicitly in the `host`:
- `setuptools`, see https://github.com/opencv/opencv/blob/4223495e6cd67011f86b8ecd9be1fa105018f3b1/modules/python/package/setup.py#L4
- `wheel`

And `py-opencv` probably has a different license - `Apache-2.0`, see https://github.com/opencv/opencv/blob/4223495e6cd67011f86b8ecd9be1fa105018f3b1/modules/python/package/setup.py#L20